logsrvd/logsrvd.c

```diff
@@ -923,12 +1300,20 @@ register_listener(struct listen_address *addr, struct sudo_event_base *base)
     debug_decl(register_listener, SUDO_DEBUG_UTIL)
 
     sock = create_listener(addr);
+
+#if defined(HAVE_OPENSSL)
+    if (logsrvd_conf_get_tls_opt() == true) {
+        struct logsrvd_tls_runtime *tls_runtime = logsrvd_get_tls_runtime();
+        if ((tls_runtime->ssl_ctx = init_tls_server_context()) == NULL)
+            sudo_fatal(NULL);
+    }
+#endif
```

Comment:
There can be multiple listen addresses but we only want to init TLS once.  This block should probably be moved after the register_listener() calls in main().

Reply:
fixed